### PR TITLE
Expose case roles endpoint for the case+user

### PR DESCRIPTION
**Pull request by mounikahmcts**
https://tools.hmcts.net/jira/browse/DIV-4632

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAccessApi.java [hunk 1]
```diff
@@ -55,4 +57,12 @@ void revokeAccessToCase(
             @PathVariable("idToDelete") final String idToDelete
     );
 
+    @PutMapping("/cases/{caseReference}/users/{userId}")
```
Comment (timja) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAccessApi.java]: Where is this endpoint in CCD?
I can't see it here:
https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/endpoint/std/CaseAccessEndpoint.java

This API maps to that endpoint really.

You may want another class for it

Comment (mounikahmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAccessApi.java]: Its available in https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/v2/external/controller/CaseUserController.java

but it is related to granting specific roles to the case similar to creator role provided granted access / revoke access. So thought of adding it here.

Do you want me to add it in a separate class?

Comment (timja) [reply src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAccessApi.java]: I think it belongs separately,
How about:
`CaseUserApi`

Comment (mounikahmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/client/CaseAccessApi.java]: > I think it belongs separately,
> How about:
> `CaseUserApi`

👍 let me change it then

**Pull request merged**